### PR TITLE
Don't allow reformat if the requested file system is unsupported

### PR DESCRIPTION
**Pull request by poncovka**
Disable reformatting on the Manual Partitioning screen if the requested
file system is not supported by Blivet or Anaconda.

In TUI, don't allow to enter the ntfs and tmpfs file systems on the Assign
Mount Points screen. We should support the same file systems as in GUI.

### CODE REVIEW

**Comment by poncovka** [comment]
@vojtechtrefny, can you have a look, please?

File: pyanaconda/modules/storage/devicetree/utils.py [hunk 1]
```diff
@@ -35,23 +35,35 @@
...
+    :param fmt_type: a name of the formatting type
+    :return: True or False
+    """
+    fmt = get_format(fmt_type)
+
+    if not fmt.type:
+        return False
+
+    # Check if the filesystem is unsupported by Anaconda.
+    if fmt.type in ("ntfs", "tmpfs"):
```
Comment (vojtechtrefny) [inline pyanaconda/modules/storage/devicetree/utils.py:50]: Btrfs was removed intentionally? It makes sense to support "btrfs as filesystem" not only as a volume manager, but it might be worth doing this in a separate commit for clarity.

**Review by vojtechtrefny — APPROVED** [review]
Looks good to me.

**Comment by jstodola** [comment]
Why should anaconda not allow to re-format a filesystem like ntfs with a different supported filesystem? Are there any known problems?

**Comment by poncovka** [comment]
> Why should anaconda not allow to re-format a filesystem like ntfs with a different supported filesystem? Are there any known problems?

I will have to reword the commit message. This check is for filesystems that will be created by the installer. It is not for the original formats. Therefore, it should be possible to re-format a filesystem like ntfs with a supported one.

File: pyanaconda/modules/storage/devicetree/utils.py [hunk 2]
```diff
@@ -35,23 +35,35 @@
 log = get_module_logger(__name__)
 
 
+def is_supported_filesystem(fmt_type):
+    """Is the filesystem supported?
+
+    :param fmt_type: a name of the formatting type
+    :return: True or False
```
Comment (VladimirSlavik) [inline pyanaconda/modules/storage/devicetree/utils.py:42]: `bool` ?

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Comment by Conan-Kudo** [comment]
I'm confused what this change is supposed to do and what it's for? Can someone explain this to me? I see it reworks how it interacts with Btrfs, so I'm concerned...

**Comment by poncovka** [comment]
I have split the code into several smaller commits to make it easier to understand. The changes are summarized in the description. Nothing else should be affected. Btrfs should work the same way as before.

File: pyanaconda/modules/storage/devicetree/utils.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/modules/storage/devicetree/utils.py]: Explained in a new commit.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me!
